dataprofiler/profilers/profile_builder.py

```diff
@@ -1420,26 +1424,32 @@ def _merge_correlation(self, other):
         if corr_mat1 is None or corr_mat2 is None:
             return None
 
-        if len(corr_mat1.columns) != len(corr_mat2.columns) or \
-                len(corr_mat1.columns) <= 1:
+        col_ids1 = np.where(~np.isnan(corr_mat1).all(axis=0))[0]
+        col_ids2 = np.where(~np.isnan(corr_mat2).all(axis=0))[0]
+
+        if len(col_ids1) != len(col_ids2) or len(col_ids1) <= 1:
```

Comment:
I'm not sure this is correct. Just because one profile has a `NaN` in one column doesn't mean it should not be mergeable for the other columns, right?

Reply:
agree. We can discuss further on that later.